homeassistant/components/wemo.py

```diff
@@ -76,6 +80,27 @@ def discovery_dispatch(service, discovery_info):
         discovery.load_platform(hass, component, DOMAIN, discovery_info,
                                 config)
 
+    def get_model_from_uuid(uuid):
+        """Determine Wemo model from UUID."""
+        if uuid is None:
+            return None
+        elif uuid.startswith('uuid:Socket'):
```

Comment:
Instead of adding another layer, you could use the updated `WEMO_MODEL_DISPATCH` dict to get the correct component name directly.
```py
def get_component_from_uuid(uuid):
    if uuid is None:
        return None
    for model, component in WEMO_MODEL_DISPATCH:
        if uuid.startswith('uuid:{}'.format(model)):
            return component
    return None
```
Then either pass the `component` or the `uuid` in the `discovery_info`, using the `get_component_from_uuid` function either before firing the discovery event or inside `discovery_dispatch`.